docs/src/main/asciidoc/rest-reactive.adoc

```diff
@@ -0,0 +1,1610 @@
+method takes parameters of the following type:
+
+.Table Context object
+|===
+|Type|Usage
+
+|link:{jaxrsapi}/javax/ws/rs/core/HttpHeaders.html[`HttpHeaders`]
+|All the request headers
+
+|link:{jaxrsapi}/javax/ws/rs/container/ResourceInfo.html[`ResourceInfo`]
+|Information about the current endpoint method and class (requires reflection)
+
+|link:{jaxrsapi}/javax/ws/rs/core/SecurityContext.html[`SecurityContext`]
+|Access to the current user and roles
+
+|`SimpleResourceInfo`
+|Information about the current endpoint method and class (no reflection required)
+
+|link:{jaxrsapi}/javax/ws/rs/core/UriInfo.html[`UriInfo`]
+|Provides information about the current endpoint and application URI
+
+|link:{jaxrsapi}/javax/ws/rs/core/Application.html[`Application`]
+|Advanced: Current JAX-RS application class
+
+|link:{jaxrsapi}/javax/ws/rs/core/Configuration.html[`Configuration`]
+|Advanced: Configuration about the deployed JAX-RS application
+
+|link:{jaxrsapi}/javax/ws/rs/ext/Providers.html[`Providers`]
+|Advanced: Runtime access to JAX-RS providers
+
+|link:{jaxrsapi}/javax/ws/rs/core/Request.html[`Request`]
+|Advanced: Access to the current HTTP method and <<preconditions>>
+
+|link:{jaxrsapi}/javax/ws/rs/core/ResourceContext.html[`ResourceContext`]
+|Advanced: access to instances of endpoints
+
+|`ServerRequestContext`
+|Advanced: RESTEasy Reactive access to the current request/response
+
+|link:{jaxrsapi}/javax/ws/rs/sse/Sse.html[`Sse`]
+|Advanced: Complex SSE use-cases
+
+|===
+
+For example, here is how you can return the name of the currently logged-in user:
+
+[source,java]
+----
+package org.acme.rest;
+
+import java.security.Principal;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.SecurityContext;
+
+@Path("user")
+public class Endpoint {
+
+    @GET
+    public String userName(SecurityContext security) {
+        Principal user = security.getUserPrincipal();
+        return user != null ? user.getName() : "<NOT LOGGED IN>";
+    }
+}
+----
+
+You can also inject those context objects using 
+https://javadoc.io/static/javax.inject/javax.inject/1/javax/inject/Inject.html[`@Inject`] on fields of the same
+type:
+
+[source,java]
+----
+package org.acme.rest;
+
+import java.security.Principal;
+
+import javax.inject.Inject;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.SecurityContext;
+
+@Path("user")
+public class Endpoint {
+
+    @Inject
+    SecurityContext security;
+
+    @GET
+    public String userName() {
+        Principal user = security.getUserPrincipal();
+        return user != null ? user.getName() : "<NOT LOGGED IN>";
+    }
+}
+----
+
+Or even on your endpoint constructor:
+
+[source,java]
+----
+package org.acme.rest;
+
+import java.security.Principal;
+
+import javax.inject.Inject;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.SecurityContext;
+
+@Path("user")
+public class Endpoint {
+
+    SecurityContext security;
+
+    @Inject
```

Comment:
`@Inject` is not necessary here

Reply:
Oh really? Cool. Didn't know that myself :)

Reply:
Done, and rendered.

Reply:
👍🏼